Compile error starting example app

Running directly in the `studio/docs/example/server` directory. Node v16.14.2, npm 8.5.0, ts-node v10.7.0

```
> npm run build

> @mathigon/studio-example@0.1.0 build
> mgon-build --assets --minify --search

  ✔ Built icons.svg in 27ms
  ✔ Built frontend/accounts.scss in 5408ms
  ✔ Built docs/example/content/geography/styles.scss in 3450ms
  ✔ Built frontend/main.scss in 4209ms
  ✔ Built docs/example/frontend/custom.scss in 3585ms
  ✔ Built docs/example/content/science/styles.scss in 3445ms
  ✔ Built frontend/dashboard.scss in 4401ms
  ✔ Built docs/example/frontend/course.scss in 4025ms
  ✔ Built docs/example/frontend/custom.ts in 5432ms
  ✔ Built frontend/main.ts in 5436ms
  ✔ Built frontend/dashboard.ts in 5442ms
  ✔ Built frontend/accounts.ts in 5458ms
  ✔ Built docs/example/content/geography/functions.ts in 3467ms
  ✔ Built docs/example/content/science/functions.ts in 3466ms
  ✔ Built docs/example/frontend/course.ts in 5444ms
  ✔ Built search-index.json and search-docs.json in 6ms
  DONE!
```
```
> npm start

> @mathigon/studio-example@0.1.0 start
> ts-node -s server/app.ts

/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:840
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
../../server/models/progress.ts:119:39 - error TS7006: Parameter 'm' implicitly has an 'any' type.

119   const messages = this.messages?.map(m => ({content: m.content, type: m.kind}));
                                          ~
../../server/models/progress.ts:180:34 - error TS7006: Parameter 'm' implicitly has an 'any' type.

180     messages: this.messages?.map(m => ({content: m.content, kind: m.kind})),
                                     ~

    at createTSError (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:840:12)
    at reportTSError (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:844:19)
    at getOutput (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:1034:36)
    at Object.compile (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:1342:43)
    at Module.m._compile (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:1474:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/aednichols/Projects/studio/docs/example/node_modules/ts-node/src/index.ts:1478:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  diagnosticCodes: [ 7006, 7006 ]
}
```

--- a/server/models/progress.ts
+++ b/server/models/progress.ts
@@ -116,7 +116,7 @@ ProgressSchema.methods.getSectionData = async function(sectionId: string) {
   }
 
   const section = sectionId ? this.sections?.get(sectionId) : undefined;
-  const messages = this.messages?.map(m => ({content: m.content, type: m.kind}));
+  const messages = this.messages?.map((m: { content: any; kind: any; }) => ({content: m.content, type: m.kind}));
   return {completed: section?.completed, activeStep: section?.activeStep, messages, steps};
 };
 
@@ -177,7 +177,7 @@ ProgressSchema.methods.getJSON = function(sectionId?: string) {
   return JSON.stringify({
     completed: section ? section.completed : undefined,
     activeStep: section ? section.activeStep : undefined,
-    messages: this.messages?.map(m => ({content: m.content, kind: m.kind})),
+    messages: this.messages?.map((m: { content: any; kind: any; }) => ({content: m.content, kind: m.kind})),
     steps
   });
 };

--- a/server/models/progress.ts
+++ b/server/models/progress.ts
@@ -108,7 +108,7 @@ ProgressSchema.virtual('activeSection').get(function(this: ProgressDocument) {
   return course.sections[p];
 });
 
-ProgressSchema.methods.getSectionData = async function(sectionId: string) {
+ProgressSchema.methods.getSectionData = async function(this: ProgressDocument, sectionId: string) {
   const steps: StepData = {};
   // TODO Only return data for steps in the requested section.
   for (const [key, step] of this.steps?.entries() || []) {
@@ -116,7 +116,7 @@ ProgressSchema.methods.getSectionData = async function(sectionId: string) {
   }
 
   const section = sectionId ? this.sections?.get(sectionId) : undefined;
-  const messages = this.messages?.map((m: { content: any; kind: any; }) => ({content: m.content, type: m.kind}));
+  const messages = this.messages?.map(m => ({content: m.content, type: m.kind}));
   return {completed: section?.completed, activeStep: section?.activeStep, messages, steps};
 };
 
@@ -165,7 +165,7 @@ ProgressSchema.methods.updateData = function(sectionId: string, changes: ChangeD
   return addedScores;
 };
 
-ProgressSchema.methods.getJSON = function(sectionId?: string) {
+ProgressSchema.methods.getJSON = function(this: ProgressDocument, sectionId?: string) {
   // TODO Only return data for steps in the requested section.
   const steps: StepData = {};
   for (const [key, data] of this.steps.entries()) {
@@ -177,7 +177,7 @@ ProgressSchema.methods.getJSON = function(sectionId?: string) {
   return JSON.stringify({
     completed: section ? section.completed : undefined,
     activeStep: section ? section.activeStep : undefined,
-    messages: this.messages?.map((m: { content: any; kind: any; }) => ({content: m.content, kind: m.kind})),
+    messages: this.messages?.map(m => ({content: m.content, kind: m.kind})),
     steps
   });
 };

